Shared/Samples/Add dynamic entity layer/AddDynamicEntityLayerView.Model.swift

```diff
@@ -0,0 +1,114 @@
+    class Model: ObservableObject {
+        /// A map with a streets basemap style.
+        let map = Map(basemapStyle: .arcGISStreets)
+        
+        /// The data source for the dynamic entity layer.
+        var streamService: ArcGISStreamService = {
+            let streamService = ArcGISStreamService(url: .streamService)
+            
+            let filter = ArcGISStreamServiceFilter()
+            filter.whereClause = "speed > 0"
+            streamService.filter = filter
+            
+            streamService.purgeOptions.maximumDuration = TimeInterval(5 * 60)
+            return streamService
+        }()
+        
+        /// The layer displaying the dynamic entities on the map.
+        var dynamicEntityLayer: DynamicEntityLayer
+        
+        /// A Boolean value indicating whether track lines should be displayed.
+        @Published var showsTrackLine: Bool {
+            didSet {
+                dynamicEntityLayer.trackDisplayProperties.showsTrackLine = showsTrackLine
+            }
+        }
+        
+        /// A Boolean value indicating whether previous observations should be displayed.
+        @Published var showsPreviousObservations: Bool {
+            didSet {
+                dynamicEntityLayer.trackDisplayProperties.showsPreviousObservations = showsPreviousObservations
+            }
+        }
+        
+        /// The maximum number of previous observations to display.
+        @Published var maximumObservations: CGFloat {
+            didSet {
+                dynamicEntityLayer.trackDisplayProperties.maximumObservations = Int(maximumObservations)
+            }
+        }
+        
+        // The maximum observations range.
+        // Used by Slider, which requires CGFloat values.
+        let maxObservationRange = CGFloat(1)...CGFloat(16)
+        
+        /// The stream service connection status.
+        @Published var connectionStatus: String
+        
+        init() {
+            // Create the dynamic entity layer
+            dynamicEntityLayer = DynamicEntityLayer(dataSource: streamService)
+            
+            // Initialize properties from the dynamic entity layer and stream service.
+            showsTrackLine = dynamicEntityLayer.trackDisplayProperties.showsTrackLine
+            showsPreviousObservations = dynamicEntityLayer.trackDisplayProperties.showsPreviousObservations
+            maximumObservations = CGFloat(dynamicEntityLayer.trackDisplayProperties.maximumObservations)
+            connectionStatus = streamService.connectionStatus.description
+            
+            Task {
+                // This will update `connectionStatus` when the stream service
+                // connection status changes.
+                for await status in streamService.$connectionStatus {
+                    DispatchQueue.main.async { [weak self] in
+                        self?.connectionStatus = status.description
+                    }
+                }
+            }
```

Comment:
While it may be convenient to kick off the connection status "observation" in the initializer, another way we usually use is to add it to the `task` modifier on the main view suggested by @philium . A few benefits

- The task will have the same lifecycle as the main view, which is the equivalent to that of the sample
- No need to dispatch onto main queue when it is UI related
- No need to explicitly manage the `Task` in the model. When you create a task block it returns a handle that you can use to cancel or get the result from it. 

To my understanding, an unstructured task will have the same lifetime as its scope at creation, which is that of `Model`. It is not a problem in this sample, but IMO the less unstructured tasks, the less risk for a memory leak.

```swift
.task {
    // This will update `connectionStatus` when the stream service
    // connection status changes.
    for await status in model.streamService.$connectionStatus {
        model.connectionStatus = status.description
    }
}
```